docs/concepts.md

```diff
@@ -50,9 +53,325 @@ Dynamic provisioning is when an resource claim does not find a matching resource
 The newly provisioned resource is automatically bound to the resource claim.
 To enable dynamic provisioning the administrator needs to create one or more resource class objects.
 
-## Connection Secrets
-
-Workloads reference all the resources the consume in their `resources` section.
+### Connection Secrets
+Workloads reference all the resources they consume in their `resources` section.
 This helps Crossplane setup connectivity between the workload and resource, and create objects that hold connection information.
 For example, for a database provisioned and managed by Crossplane, a secret will be created that contains a connection string, user and password.
 This secret will be propagated to the target cluster so that it can be used by the workload.
+
+### Secure Connectivity
+networks, subnets, peering, NACLs,etc.
+
+# Glossary
+
+## Kubernetes
+Crossplane is built on the Kubernetes API machinery as a platform for declarative management,
+and common terminology used from the [Kubernetes Glossary][kubernetes-glossary] where possible,
+and we don't seek to reproduce that glossary here.
+
+[kubernetes-glossary]: https://kubernetes.io/docs/reference/glossary/?all=true
+However we'll summarize some commonly used concepts for convenience.
+
+### CRD
+A standard Kubernetes Custom Resource Definition (CRD), which defines a new type of resource that can be managed declaratively. 
+This serves as the unit of management in Crossplane.
+The CRD is composed of spec and status sections and supports API level versioning (e.g., v1alpha1)
+
+### Controller
+A standard Kubernetes Custom Controller, providing active control loops that own or more CRDs. 
+Can be implemented in different ways, such as 
+golang code (controller-runtime), templates, functions/hooks, templates, a new DSL, etc. 
+The implementation itself is versioned using semantic versioning (e.g., v1.0.4)
+
+### Namespace
+Allows logical grouping of resources in Kubernetes that can be secured with RBAC rules.
+
+## Crossplane
+### External CRD
+Usually represent an external resource, and cannot be broken down any further (leaves)
+
+aka Atomic CRD
+
+### Composite CRD
+A resource that captures parent/child relationships. They have a selector that can help query/find children resources.
+
+### Claim CRD
+An abstract resource that binds to managed resources, like a MySQLInstance..
+
+### App CRD
+An abstract resource that manages an application instance, like a WordpressInstance.
```

Comment:
Here and in other cases we seem to be using "CRD" as an _instance of the resource_ rather than as an _instance of the definition of the resource_. Anything of `kind: MySQLInstance` is a (custom) resource, not a CRD.

Reply:
removed